### PR TITLE
MinCED wrapper

### PR DESCRIPTION
**Pull request by tkosciol**
I added MinCED wrapper and created tests for it.

### CODE REVIEW

File: micronota/bfillings/tests/test_minced.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            prefix = self.positive_prefix
+            res = predict_crispr(fp, self.temp_dir,
+                                 prefix, params,
+                                 gff=flags['gff'],
+                                 gffFull=flags['gffFull'],
+                                 spac=flags['spacers'])
+            self.assertEqual(res['ExitStatus'], 0)
+            # get output from flags
+            # for suffix in [k for k, v in flags.items()
+            #                if v is True and k is not 'spacers']:
```
Comment (mortonjt) [inline micronota/bfillings/tests/test_minced.py]: We don't need these lines anymore do we?

**Comment by mortonjt** [comment]
:+1: Don't forget to update the [CHANGELOG](https://github.com/biocore/micronota/blob/master/CHANGELOG.md) to mention that we now have the minced wrapper.

@RNAer want to review this before merging?

File: micronota/bfillings/minced.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+    if params is None:
+        params = {}
+
+    if gff:
+        out_suffix = 'gff'
+    elif gffFull:
+        out_suffix = 'gffFull'
+    else:
+        out_suffix = 'crisprs'
```
Comment (RNAer) [inline micronota/bfillings/minced.py]: what if multiple of them are `True`?

File: micronota/bfillings/minced.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            # replace the final extension with '_spacers.fa'
+            out_fp = re.sub(r"\.[^.]*$", '_spacers.fa', data[1])
+            result['spacers'] = ResultPath(Path=out_fp, IsWritten=True)
+
+        return result
+
+
+def predict_crispr(in_fp, out_dir, prefix, params=None,
+                   spac=False, gff=False, gffFull=False):
+    '''Predict CRISPRs for the input file.
```
Comment (RNAer) [inline micronota/bfillings/minced.py]: I would put `params` at the very end.

**Comment by RNAer** [comment]
Thanks, @tkosciol ! It looks good. Besides the minor comments inline, why are there changes in the files of hmmer and infernal folder?

File: micronota/bfillings/minced.py (hunk 2 above)
Comment (tkosciol) [inline micronota/bfillings/minced.py]: The way that MinCED works is that if there is a -gff flag, it supresses the full output. There can be a case where one uses -gff and -gffFull and then we get 2 files, but a gff is a shorter version of gffFull, so I don't think it's a good practice to use both flags.

Comment (RNAer) [inline micronota/bfillings/minced.py]: that makes sense. but the `crisprs` output should be able to be enabled
together with gff/gfffull.

Comment (tkosciol) [inline micronota/bfillings/minced.py]: I don't know if I understand you correctly. The way MinCED is written:
If minced -gff INP OUT: OUT is a gff file
If minced INP OUT: OUT is a crisprs file
If minced -gffFull INP OUT: OUT is a gffFull file
If minced -spacers -gff INP OUT: there are *2* output files OUT.gff and OUT_spacers.gff

So there is no way to enable both gff and crisprs files in one run.

**Comment by tkosciol** [comment]
@RNAer thanks! About changes in hmmer and infernal - I don't know how they got there. There is just some version issue between my repo and micronota master. It's just that I don't know how to remove those files selectively from a pull request, sorry.:/

**Comment by mortonjt** [comment]
Those files were automagically merged in from the previous PR.  Given that the tests are passing, I think it is OK.

File: micronota/bfillings/minced.py (hunk 2 above)
Comment (RNAer) [inline micronota/bfillings/minced.py]: ah sorry, I misunderstood. you're right.

**Comment by RNAer** [comment]
nope. those files are never tracked before. I don't think we should add random files.

**Comment by RNAer** [comment]
:+1:  Thanks, @tkosciol !

**Pull request merged**